ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -37,10 +41,33 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = signed_address_change;
+
+    assert!(address_change.message.validator_index < state.validators.len());
+
+    let validator = &mut state.validators[address_change.message.validator_index];
+
+    assert!(validator.withdrawal_credentials.starts_with(&[BLS_WITHDRAWAL_PREFIX]));
+    assert!(
+        validator.withdrawal_credentials[1..]
+            == hash(address_change.message.from_bls_public_key.as_ref())[1..]
+    );
+
+    let domain =
+        compute_domain(BlsToExecutionChange, None, Some(state.genesis_validators_root), context)?;
```

Comment:
nitpick but I prefer to grab these from the enum
```suggestion
        compute_domain(DomainType::BlsToExecutionChange, None, Some(state.genesis_validators_root), context)?;
```

Reply:
done